src/pat/inject/inject.js

```diff
@@ -305,6 +314,7 @@ define([
             if ((cfg.history === "record") && ("pushState" in history)) {
                 history.pushState({'url': cfg.url}, "", cfg.url);
             }
+            $target.trigger("pat-inject-content-loaded-final");
         },
```

Comment:
> could i use imagesLoaded jquery plugin when there are images to trigger only one `pat-inject-content-loaded` event? and in case there is no images we only trigger one `pat-inject-content-loaded` event as well.

Yeah, that's roughly what I had in mind as well, but I looked at the code a bit closer now and see another potential issue.

Triggering an event on `$target` won't work in all cases, because `$target` might be replaced entirely by a new element and might therefore not be in the DOM anymore.

In the `_afterInjection` method the `patterns-injected` event is fired. That looks to me like the right event for you to listen to instead. Agreed?